### PR TITLE
Merge offline feature

### PR DESCRIPTION
**Pull request by haijian-vaadin**
Merge the docs for offline features to master.

### CODE REVIEW

**Comment by jouni** [comment]
Gah, now Vale fails on some new error again. Something related to `custom-connection-indicator.ts`.

**Comment by jouni** [comment]
The preview deployment fails on

```
12:51:49 PM: ERROR in [at-loader] ./frontend/demo/fusion/forms/contact-form.ts:12:31
12:51:49 PM:     TS2307: Cannot find module 'promise-file-reader' or its corresponding type declarations.
```

**Comment by joheriks** [comment]
> The preview deployment fails on
> 
> ```
> 12:51:49 PM: ERROR in [at-loader] ./frontend/demo/fusion/forms/contact-form.ts:12:31
> 12:51:49 PM:     TS2307: Cannot find module 'promise-file-reader' or its corresponding type declarations.
> ```

We need to add `promise-file-reader` as a dependency (possibly worked previously due to being a transitive dependency of some other package). I assume it must be a runtime dependency (not dev) for this check to pass?

**Comment by jouni** [comment]
> I assume it must be a runtime dependency (not dev) for this check to pass?

All files under `frontend` will be part of the docs-app build, so they need to compile.

**Comment by haijian-vaadin** [comment]
My bad, I forgot the `promise-file-reader` dependency when resetting the `package.json` file. Now it's all green.

File: package.json [hunk 1]
```diff
@@ -47,7 +47,8 @@
       "@vaadin/vaadin-material-styles": "1.3.2",
       "@vaadin/vaadin-rich-text-editor": "1.3.0",
       "lit-element": "2.3.1",
-      "@vaadin/vaadin-avatar": "1.0.2"
+      "@vaadin/vaadin-avatar": "1.0.2",
+      "promise-file-reader": "^1.0.3"
```
Comment (joheriks) [inline package.json:51]: Should be added to `/dependencies` rather than `vaadin/dependencies`?

**Review by joheriks — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (haijian-vaadin) [reply package.json]: done

File: articles/fusion/forms/appendix-form-images.asciidoc [hunk 2]
```diff
@@ -0,0 +1,45 @@
+---
+title: Appendix: Image Fields in Client-Side Forms
```
Comment (joheriks) [inline articles/fusion/forms/appendix-form-images.asciidoc:2]: This probably should also have prefix "Appendix" in the sidebar, as it has been removed from other subsections that previously had it (e.g. "Appendix: Binding Data to Vaadin Components").

Comment (haijian-vaadin) [reply articles/fusion/forms/appendix-form-images.asciidoc]: done

**Review by joheriks — APPROVED** [review]

**Pull request merged**